### PR TITLE
Rasing click.Abort to not stop the workflow

### PR DESCRIPTION
**Pull request by karlnyr**
This PR aims to keep the command `cg workflow mip-dna store completed` going even though one sample fails to be stored. This fix would allow the iteration over all completed cases to finish, despite one breaking.

### How to prepare for test
- [x] ssh to hasta
- [x] install on stage:
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh fix/completed-abort-fix`

### How to test
- [x] login to stage
- [x] do `cg workflow mip-dna store analysis path/to/case_config.yaml`
- [x] do `cg workflow mip-dna store completed`

### Expected test outcome
- [x] Analysis can be stored in Housekeeper
- [x] Completed continues despite errors
- [x] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] code approved by
- [x] tests executed by @karlnyr 
- [ ] "Merge and deploy" approved by

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by karlnyr** [comment]
Installing on hasta:

image

When running `cg workflow mip-dna store analysis vitalmouse_config.yaml` the files should be included in Housekeeper:
image

When running `cg workflow mip-dna store completed choicehusky should fail due to missing mandatory files:
image

... and vitalmouse should be saved:
image

**Comment by Mropat** [comment]
Beautiful code and amazing job with the tests!!  :fire: :fire: :fire:

**Review by Mropat — APPROVED** [review]

File: cg/cli/workflow/mip/store.py [hunk 1]
```diff
@@ -38,9 +37,10 @@ def analysis(context, config_stream):
     status = context.obj["db"]
     hk_api = context.obj["hk_api"]
 
+    exit_code = EXIT_SUCCESS
     if not config_stream:
         LOG.error("Provide a config file.")
-        context.abort()
+        raise click.Abort()
```
Comment (moonso) [inline cg/cli/workflow/mip/store.py:43]:
```suggestion
        raise click.Abort
```

File: cg/cli/workflow/mip/store.py [hunk 2]
```diff
@@ -57,11 +57,13 @@ def analysis(context, config_stream):
         MandatoryFilesMissing,
     ) as error:
         click.echo(click.style(error.message, fg="red"))
-        context.abort()
+        exit_code = EXIT_FAIL
     except FileNotFoundError as error:
+        print(error)
```
Comment (moonso) [inline cg/cli/workflow/mip/store.py:62]:
```suggestion
```
?

File: cg/cli/workflow/mip/store.py [hunk 3]
```diff
@@ -57,11 +57,13 @@ def analysis(context, config_stream):
...
         click.echo(click.style(f"missing file: {error.args[0]}", fg="red"))
-        context.abort()
-
+        exit_code = EXIT_FAIL
+    if exit_code:
+        raise click.Abort()
```
Comment (moonso) [inline cg/cli/workflow/mip/store.py:66]:
```suggestion
        raise click.Abort
```

File: cg/cli/workflow/mip/store.py [hunk 4]
```diff
@@ -83,8 +85,8 @@ def completed(context):
...
                 context.invoke(analysis, config_stream=config_stream)
-            except Exception:
+            except (Exception, click.Abort):
                 LOG.error("case storage failed: %s", analysis_obj.family, exc_info=True)
-                exit_code = FAIL
-
-    sys.exit(exit_code)
+                exit_code = EXIT_FAIL
+    if exit_code:
+        raise click.Abort()
```
Comment (moonso) [inline cg/cli/workflow/mip/store.py:92]:
```suggestion
        raise click.Abort
```

File: tests/cli/workflow/mip/conftest.py [hunk 5]
```diff
@@ -15,6 +27,209 @@ def mock_root_folder():
     return "/var/empty"
 
 
+@pytest.fixture(name="mip_case_ids")
+def mip_case_ids() -> dict:
```
Comment (moonso) [inline tests/cli/workflow/mip/conftest.py:31]:
```suggestion
def fixture_mip_case_ids() -> dict:
```

File: tests/cli/workflow/mip/conftest.py [hunk 6]
```diff
@@ -15,6 +27,209 @@ def mock_root_folder():
...
+    for case in mip_case_ids:
+        case_dir = Path(project_dir / "cases" / case)
+        case_dir.mkdir(exist_ok=True, parents=True)
+        mip_case_dirs[case] = case_dir
+
+    return mip_case_dirs
+
+
+@pytest.fixture(name="mip_deliverables")
+def make_mip_deliverables(
```
Comment (moonso) [inline tests/cli/workflow/mip/conftest.py:70]:
```suggestion
def fixture_make_mip_deliverables(
```
Although base on the name it sounds like this should rather be a function than a fixture, right?

File: tests/cli/workflow/mip/conftest.py [hunk 7]
```diff
@@ -15,6 +27,209 @@ def mock_root_folder():
...
+            case_deliverables_path = mip_case_dirs[case] / f"{case}_deliverables.yaml"
+            with open(case_deliverables_path, "w") as fh:
+                yaml.dump({"files": []}, fh)
+            deliverables_paths[case] = case_deliverables_path
+
+    return deliverables_paths
+
+
+@pytest.fixture(name="mip_qc_sample_info")
+def make_mip_sample_infos(mip_case_ids: dict, mip_case_dirs: dict) -> dict:
```
Comment (moonso) [inline tests/cli/workflow/mip/conftest.py:114]: Same like above

File: tests/cli/workflow/mip/conftest.py [hunk 8]
```diff
@@ -15,6 +27,209 @@ def mock_root_folder():
...
+        sample_info_path = mip_case_dirs[case] / f"{case}_qc_sample_info.yaml"
+        qc_sample_info[case] = sample_info_path
+        with open(sample_info_path, "w") as fh:
+            yaml.dump(sample_info_data, fh)
+
+    return qc_sample_info
+
+
+@pytest.fixture(name="mip_configs")
+def make_mip_configs(
```
Comment (moonso) [inline tests/cli/workflow/mip/conftest.py:141]: Same

File: tests/cli/workflow/mip/conftest.py [hunk 9]
```diff
@@ -15,6 +27,209 @@ def mock_root_folder():
...
+@pytest.fixture(name="empty_housekeeper_api")
+def populate_mip_housekeeper(real_housekeeper_api: HousekeeperAPI) -> HousekeeperAPI:
+    """Empty housekeeper for testing store in mip workflow"""
+
+    _housekeeper_api = real_housekeeper_api
+
+    return _housekeeper_api
+
+
+@pytest.fixture(name="_tb_api")
```
Comment (moonso) [inline tests/cli/workflow/mip/conftest.py:215]:
```suggestion
@pytest.fixture(name="tb_api")
```
Could you give this a name that tells more about what it is? Perhaps `populated_tb_api` or something

**Review by moonso — CHANGES_REQUESTED** [review]
Wow great job @karlnyr 🦸 😸 !

File: tests/cli/workflow/mip/conftest.py (hunk 6 above)
Comment (karlnyr) [reply tests/cli/workflow/mip/conftest.py:70]: I think its better to call it fixture_mip_deliverables, I will change it for the other ones as well in another commit

File: tests/cli/workflow/mip/conftest.py (hunk 9 above)
Comment (karlnyr) [reply tests/cli/workflow/mip/conftest.py:215]: Will do!

**Review by moonso — APPROVED** [review]
👍

**Comment by Mropat** [comment]
Deployed on stage
image

Deployed on prod
image

**Pull request merged**